core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -134,12 +136,29 @@ object SparkSubmit {
       }
     }
 
+    val isYarnCluster = clusterManager == YARN && deployMode == CLUSTER
+
+    // Require all python files to be local, so we can add them to the PYTHONPATH
+    // when yarn-cluster, all python files can be non-local
+    if (args.isPython && !isYarnCluster) {
+      if (Utils.nonLocalPaths(args.primaryResource).nonEmpty) {
+        SparkSubmit.printErrorAndExit(
```

Comment:
if we move it to SparkSubmitArguments, we need to get clusterManager and deployMode. But this work has been did in SparkSubmit. there are some repeated works.